Chessington.GameEngine/Square.cs

```diff
@@ -49,5 +51,22 @@ public override string ToString()
         {
             return string.Format("Row {0}, Col {1}", Row, Col);
         }
+
+        public bool IsInbound()
+        {
+            return Enumerable.Range(0, 8).Contains(this.Col) && Enumerable.Range(0, 8).Contains(this.Row);
```

Comment:
8 and similar numbers in a middle of code are called "Magic numbers", You should assign in to a constant and reuse it. 
In our case it's already a constant (refer to GameSettings class)